src/cfnlint/core.py

```diff
@@ -248,6 +265,15 @@ def get_template_args_rules(cli_args):
 
     args = parser.parse_args(cli_args)
 
+    # If the template has cfn-lint Metadata but the same options are set on the command-
```

Comment:
you can specify configuration in the Metadata section of the CloudFormation template.  This would allow someone to ignore rules, and such right in the template.  CLI parameters will always win.  this may have to be approached differently now with teh changes you are making.  I may be able to take a swing at that so you can focus on the other part.

Reply:
now I'm a little bit confused as I am reading this more.  Is this some prework for a future change?  https://github.com/awslabs/cfn-python-lint/blob/master/src/cfnlint/core.py#L250-L252 will set the defaults and then reparse the arguments to see if they should be overridden.  This may not be the best way to do that but I'm wondering if this is duplicative and should result in that old code getting deleted.

Reply:
No, this PR is completely standalone. Without this block, what you get is that the template exclusion and the commandline exclusion are both active. This is because the new argparse action is "extend", instead of the old (implicit) "replace".

A smarter solution here would be a bit of a rework of how the logic works. In current master the logic is like this:

1. Load template overrides as `template_defaults`
1. Set template overrides as CLI defaults
1. Parse CLI as `args`

The logic I've implemented is:

1. Load template overrides as `template_defaults`
1. Set template overrides as CLI defaults
1. Parse CLI as `args`
1. For every override in template_defaults:
   1. If the version in `args` starts with the override:
      1. Delete the override from the start of the `args` entry

I propose something like:

1. Load template overrides as `template_defaults`
1. Parse CLI as `args`
1. For every override setting
   1. If the CLI doesn't specify it
      1. Set it in `args`

Reply:
I get it now.  I like your proposed idea.